elastic/kibana
# [Security Solutions][Detection Engine] Creates an autocomplete package and moves duplicate code between lists and security_solution there

## Summary

Fixes https://github.com/elastic/kibana/issues/105378

Creates an autocomplete package from `lists` and removes duplicate code between `lists` and `security_solutions`
* Consolidates different PR's where we were changing different parts of autocomplete in different ways.
* Existing Cypress tests should cover any mistakes hopefully

Manual Testing:
* Ensure this bug does not crop up again https://github.com/elastic/kibana/pull/87004
* Make sure that the exception list autocomplete looks alright

### Checklist
